### DISCUSSION TITLE
thermostat go to sleep after no heater power

### DISCUSSION THREAD

**Discussion by pacoso81** [arendst/Tasmota#13512] [Q&A] [answered]
Hi.
I have a setup of 2 tasmotas and HA acting as water heater (boiler) controller.
The 1-st node is wifi wall switch and is controlling the main power to the heater.
2-nd node is my thermostat and the output is connected in series whit the output of the 1-st to control the contactor for the heaters.
The system runs independent from HA  (HA is only for setpoint and status of the thermostat and for controlling the switch).

My problem is that after i disable the power to the heaters the thermostat will enable the output but nothing is happening because of no power to the heaters and after some time it turns OFF and no longer reacts when i power up the heaters.
I have to restart the device to start working again.
I think this is because of TimeRampupMaxSet = 960 or TimeRampupCycleSet = 30, and after this time the thermostat no longer reacts.

Thermostat ControllerModeSet  is 2 = Predictive ramp-up controller
Sensor is directly connected to the device.

I was thinking to implement some mqtt switch to disable the thermostat when 1-st node is OFF and to enable when is ON.
The mqtt command must come from the 1-st node and independent from the HA (is it possible to pass the message from 1-st to 2-nd client and jump the HA mqtt broker?)
I guest that i need rules in the 1-st node to publish mqtt command and on the 2-nd node rule and subscription to receive the message and disable the thermostat.
I need help regarding this mater.

Also i'm not sure if the command ThermostatModeSet 0/1 will work for my setup, will the thermostat start and stop without restarting the device?

**Comment by ascillato** [comment]
Hi, as you didn't have any answer here, you can try addressing this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4).

**Comment by barbudor** [comment, marked as answer]
> is it possible to pass the message from 1-st to 2-nd client and jump the HA mqtt broker?

You don't need to bypass mqtt
In rules you can use 'publish' command
For example 'on power1#state=0 do publish cmnd/yourthermostat/Thermostatmodeset 0 endon' would send the command to the thermostat when power is switched off.

Hope this helps.
I have no experience with thermostat

**Comment by pacoso81** [comment]
i need to subscribe to topic in order to get the command in the mem2 and i already have subscription for mem1
how to add second subscription?
here is my Rule 1:

`Rule1`
`ON mqtt#connected DO Subscribe BoilerSet, cmnd/Perea_boiler_thermostat/TEMPTARGETSET ENDON`
`ON Event#BoilerSet DO Backlog mem1 %value% ENDON`
`ON mem1#state DO temptargetset %value% ENDON`
`ON Event#ThermostatModeSet DO Backlog mem2 %value% ENDON`
`ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;temptargetset %mem1%; ThermostatModeSet %mem2% ENDON`
`ON Time#Minute DO publish stat/Perea_boiler_thermostat/setpoint %mem1% ENDON`
`ON Power1#State=1 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"ON"} ENDON`
`ON Power1#State=0 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"OFF"} ENDON`

mem1 is for setpoint
mem2 will be for ThermostatModeSet 
but i cant get the event without subscription nor state of ThermostatModeSet 

this is what i get in console when node receive the command:
`MQT: stat/Perea_boiler_thermostat/THERMOSTATMODESET = {"ThermostatModeSet":1}`
`MQT: stat/Perea_boiler_thermostat/THERMOSTATMODESET = {"ThermostatModeSet":0}`

**Reply by sfromis** [reply]
IIRC, you can have more than one subscribe. 

However, I do not think that you at all need to use subscribe here, as I'd expect a response with a JSON payload similar to:
`{"TempTargetSet":25}`
which allows you to use a rule like 
`on TempTargetSet#data do Mem1 %value% endon`
The key here is to match exactly what you see in the JSON.

**Reply by pacoso81** [reply]
i can get the command no problem, i needed to be in memory because thermostat don't remember the last state and after restart i need to get the last state out of memory.

**Reply by pacoso81** [reply]
Finally i got it working

`Rule1 `
`ON mqtt#connected DO Backlog Subscribe THModeSet, cmnd/Perea_boiler_thermostat/ThermostatModeSet; Subscribe BoilerSet, cmnd/Perea_boiler_thermostat/TEMPTARGETSET  ENDON`
`ON Event#BoilerSet DO Backlog mem1 %value% ENDON`
`ON mem1#state DO temptargetset %value% ENDON`
`ON Event#THModeSet DO Backlog mem2 %value% ENDON`
`ON mem2#state DO ThermostatModeSet %value% ENDON`
`ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;temptargetset %mem1%; ThermostatModeSet %mem2% ENDON`
`ON Time#Minute DO publish stat/Perea_boiler_thermostat/setpoint %mem1% ENDON`
`ON Power1#State=1 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"ON"} ENDON`
`ON Power1#State=0 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"OFF"} ENDON
`

**Reply by sfromis** [reply]
Great that you got it to work. You'd probably have had less trouble if you skipped subscribe, and used my suggestion to save the value in Mem each time.

**Reply by pacoso81** [reply]
I forgot to mention that i have very little experience in JSON.
So i have tested your version and i like it, simple and efficient. 
Thank you for your help. 

`Rule1`
`ON TempTargetSet#data do Mem1 %value% ENDON`
`ON ThermostatModeSet#data DO mem2 %value% ENDON`
`ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;temptargetset %mem1%; ThermostatModeSet %mem2% ENDON`
`ON Time#Minute DO publish stat/Perea_boiler_thermostat/setpoint %mem1% ENDON`
`ON Power1#State=1 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"ON"} ENDON`
`ON Power1#State=0 DO publish tele/Perea_boiler_thermostat/STATE {"POWER":"OFF"} ENDON`

**Reply by sfromis** [reply]
Beware that if you change Mem variables many times a day, that the flash writes needed counts against lifetime of the flash chip.

**Reply by pacoso81** [reply]
Yes, i'm aware of that.
In the worst case, i have a lot of chips.
